core/spec/models/spree/store_spec.rb

```diff
@@ -1,6 +1,36 @@
 require 'spec_helper'
 
 describe Spree::Store, type: :model do
+  describe '#destroy' do
+    let!(:default_store) { create(:store, default: true) }
+    let!(:store) { create(:store, default: false) }
+
+    before { create(:store, default: true) }
+
+    it 'is soft deleted' do
+      store.destroy!
+      expect(store.deleted_at).not_to be_nil
+    end
+
+    context 'there is a store with the shared products' do
+      let(:store2) { create(:store) }
+      let(:shared_product) { build(:product) }
+      let(:not_shared_product) { build(:product) }
+
+      before do
+        Spree::Config[:disable_store_presence_validation] = true
```

Comment:
@tomurb we cannot just disable it, you need to fix https://github.com/spree/spree/blob/master/core/app/models/spree/store_product.rb#L5 like this: https://github.com/spree/spree/blob/master/core/app/models/spree/line_item.rb#L7 (`with_deleted`)

Reply:
@damianlegawiec I have no idea what to do about it.

- if I remove `Spree::Config[:disable_store_presence_validation] = true` there’s
  ```
  ActiveRecord::RecordInvalid:
            Validation failed: Stores Must have at least one Store associated
   ```
- if then I change the products’ let variables to use “create” instead of “build”, the tests are failing.
- both cases are made with `belongs_to :store, -> { with_deleted }, class_name: 'Spree::Store', touch: true in the Spree::StoreProduct`
- same if I add `-> { with_deleted }` everywhere where there’s `belongs_to :store,`

Reply:
@tomurb Try creating products _with_ stores. E.g.
```
let(:shared_product) { create(:product, stores: [store, store2]) }
let(:not_shared_product) { create(:product, stores: [store]) }
```